main.rb

```diff
@@ -58,37 +62,44 @@ def my_select
     end
   end
 
-  def my_all?
+  def my_all?(opt = nil)
     bool_item = true
-    if block_given?
-      my_each do |item|
-        bool_item = false unless yield(item)
-      end
+    if opt.nil? && block_given?
+      my_each { |item| bool_item = false unless yield(item) }
+    elsif opt.is_a?(Regexp)
+      my_each { |item| bool_item = false unless item.match(opt) }
+    elsif opt.is_a?(Module)
+      my_each { |item| bool_item = false unless item.is_a?(opt) }
+    else
+      my_each { |item| bool_item = false unless item.nil? || item == false }
     end
     bool_item
   end
```

Comment:
There's a common issue with `my_all?`, `my_any?` and `my_none?`

1. the `bool_item` should only be false **if** item is `nil` or `false`. Let's not use `unless` here.

   Test with the following cases

   ```ruby
   irb(main):002:0> [1,2,3].all?
   #=> true
   irb(main):003:0> [1,2,3].my_all?
   #=> false

   ```

   Suggested solution

   ```diff
   -      my_each { |item| bool_item = false unless item.nil? || item == false }
   +      my_each { |item| bool_item = false if item.nil? || item == false }
   ```

2. Let's handle cases where `opt` is not `nil` and no `block` is given. See example of this case below:

   ```ruby
   irb(main):006:0> [2,2,2].all?(2)
   #=> true
   irb(main):007:0> [2,2,2].my_all?(2)
   #=> false
   ```

   Note that (1) is for when `opt` is `nil`, I would recommend adding one more condition for when `opt.nil?` then use the last `else` case for case (2) above.

### Find more details about [`all?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F)

Please let's do this for both `my_any?` and `my_none?`